app/src/lib/title-bar-config.ts

```diff
@@ -0,0 +1,38 @@
+import { writeFile } from 'fs/promises'
+import { app } from 'electron'
+import { join } from 'path'
+import { TitleBarStyle } from '../ui/lib/title-bar-style'
+import { readFileSync } from 'fs'
+
+export type TitleBarConfig = {
+  titleBarStyle: TitleBarStyle
+}
+
+const getTitleBarConfigPath = () =>
+  join(app.getPath('userData'), '.title-bar-config')
+
+// The function has to be synchronous since we need its return value to create electron BrowserWindow
+function readTitleBarConfigFile(): TitleBarConfig {
+  let titleBarConfig: TitleBarConfig
+
+  try {
+    titleBarConfig = JSON.parse(readFileSync(getTitleBarConfigPath(), 'utf8'))
+
+    // Return if we found valid values
+    if (['native', 'custom'].includes(titleBarConfig.titleBarStyle)) {
+      return titleBarConfig
+    }
+  } catch (error) {}
+
+  // If the config file doesn't exists, or if it is corrupted, rewrite it with the defaults
+  titleBarConfig = { titleBarStyle: 'native' }
+  saveTitleBarConfigFile(titleBarConfig)
```

Comment:
:art: I'm still pondering whether there's a value here for persisting this config for everyone, but I could see a case for skipping this

Reply:
OK.